src/NuGet.Clients/PackageManagement.UI/Services/NuGetLockService.cs

```diff
@@ -18,8 +20,44 @@ public sealed class NuGetLockService : INuGetLockService, IDisposable
     {
         private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
 
+        private readonly JoinableTaskCollection _joinableCollection;
+        private readonly JoinableTaskFactory _joinableFactory;
+
         public bool IsLockHeld => _semaphore.CurrentCount == 0;
 
+        public NuGetLockService()
+        {
+            var joinableTaskContextNode = new JoinableTaskContextNode(ThreadHelper.JoinableTaskContext);
```

Comment:
You shouldn't need to wrap `ThreadHelper.JoinableTaskContext` in a `JoinableTaskContextNode`. The original `JoinableTaskContext` itself has the methods on it that you need.